Google.PowerShell/Compute/GceMetadataCmdlet.cs

```diff
@@ -0,0 +1,173 @@
+    /// <para>Gets the value of the custom metadata with key "customKey" placed in the project .</para>
+    /// </example>
+    /// <example>
+    /// <code>PS C:\> $metadata, $etag = Get-GceMetadata -AppendETag -Recursive</code>
+    /// <para>Gets the entire metadata tree, and the ETag of the version retrieved.</para>
+    /// </example>
+    /// <example>
+    /// <code>
+    ///     PS C:\> $newTags, $newEtag = Get-GceMetadata -Path "instance/tags" -AppendETag -WaitUpdate `
+    ///                                     -LastETag $oldETag
+    /// </code>
+    /// <para>Waits for the metadata "instance/tags" to be updated by the server.</para>
+    /// </example>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GceMetadata")]
+    [OutputType(typeof(string))]
+    public class GetGceMetadataCmdlet : GCloudCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The path to the specific metadata you wish to get e.g. "instance/tags", "instance/attributes",
+        /// "project/attributes/sshKeys".
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public string Path { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// If set, will get the metadata subtree as a JSON string. If -Path is not set, will get the entire
+        /// metadata tree as a JSON string.
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public SwitchParameter Recursive { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        ///   When set, the value of the respone ETag will be appended to the output pipeline after the content.
+        /// </para>
+        /// <para type="description">
+        /// <code>PS C:\> $metadata, $etag = Get-GceMetadata -AppendETag -Recursive</code>
+        /// </para>
+        /// <para type="description">
+        ///   Gets the entire metadata tree, and the ETag of the version retrieved.
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public SwitchParameter AppendETag { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// If true, the query will wait for the metadata to update.
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public SwitchParameter WaitUpdate { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The last ETag known. Used in conjunction with -WaitUpdate. If the last ETag does not match the
+        /// current ETag of the metadata server, it will return immediatly.
+        /// </para>
+        /// </summary>
+        [Parameter]
+        public string LastETag { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Used in conjunction with -WaitUpdate. The amout of time to wait. If the timeout expires, the 
+        /// current metadata will be returned.
```

Comment:
Maybe add another "description" block on how to know if the timeout was hit:

You can determine if an update was received or not by checking the ETag value. See the `-AppendETag` parameter.